libs/src/api/serviceProvider.ts

```diff
@@ -24,44 +26,47 @@ const CONTENT_NODE_SELECTION_EQUIVALENCY_DELTA = 200
 class ServiceProvider extends Base {
   /* ------- Content Node  ------- */
 
-  async listCreatorNodes () {
-    return this.ethContracts.ServiceProviderFactoryClient.getServiceProviderList(CONTENT_NODE_SERVICE_NAME)
+  async listCreatorNodes() {
+    return await this.ethContracts.ServiceProviderFactoryClient.getServiceProviderList(
+      CONTENT_NODE_SERVICE_NAME
+    )
   }
 
   /**
    * Fetches healthy Content Nodes filtered down to a given whitelist and blacklist
-   * @param {Set<string>?} whitelist whether or not to include only specified nodes (default no whiltelist)
-   * @param {Set<string?} blacklist whether or not to exclude any nodes (default no blacklist)
    */
-  async getSelectableCreatorNodes (
-    whitelist = null,
-    blacklist = null,
+  async getSelectableCreatorNodes(
+    whitelist: Set<string> | null = null, // whether or not to include only specified nodes (default no whiltelist)
+    blacklist: Set<string> | null = null, // whether or not to exclude any nodes (default no blacklist)
     timeout = CONTENT_NODE_DEFAULT_SELECTION_TIMEOUT
   ) {
     let creatorNodes = await this.listCreatorNodes()
 
     // Filter whitelist
     if (whitelist) {
-      creatorNodes = creatorNodes.filter(node => whitelist.has(node.endpoint))
+      creatorNodes = creatorNodes.filter((node) => whitelist.has(node.endpoint))
     }
     // Filter blacklist
     if (blacklist) {
-      creatorNodes = creatorNodes.filter(node => !blacklist.has(node.endpoint))
+      creatorNodes = creatorNodes.filter(
+        (node) => !blacklist.has(node.endpoint)
+      )
     }
 
     // Time requests and get version info
     const timings = await timeRequests({
-      requests: creatorNodes.map(node => ({
+      requests: creatorNodes.map((node) => ({
         id: node.endpoint,
         url: `${node.endpoint}/health_check/verbose`
       })),
       sortByVersion: true,
       timeout
     })
 
-    const services = {}
-    timings.forEach(timing => {
-      if (timing.response) services[timing.request.id] = timing.response.data.data
+    const services: { [id: string]: any } = {}
```

Comment:
just curious if data.data is typed in a way that helps here?

Reply:
unfortunately not